### PR TITLE
Don't use String[] args login

### PR DESCRIPTION
**Pull request by dominikl**
The login via `LoginCredentials(String[] args)` fails when the credentials contain `#` characters (which is an issue with https://doc.zeroc.com/display/Ice36/The+Properties+Interface treating '#' as comment symbol, apparently). 
Removed that from the example in favor of the `LoginCredentials(String username, String password, String host, int port)` way to login.

See also: https://trello.com/c/7LQRbxQX/86-minimal-client-gateway-and-passwords

**Test**
- Create a user with a password which contains a '#'
- Checkout the repository and run maven `mvn install`
- Then run the `SimpleConnection.java` example with command line arguments `--omero.host=HOST --omero.port=PORT --omero.user=USER --omero.pass="PASSWORD"`
- There's no output expected

### CODE REVIEW

File: src/main/java/com/example/SimpleConnection.java [hunk 1]
```diff
@@ -52,20 +52,6 @@ private void connect(String hostname, int port, String userName, String password
...
-    /** 
-     * Creates a connection, the gateway will take care of the services
-     * life-cycle.
-     *
-     * @param args The arguments used to connect.
-     */
-    private void connect(String[] args)
-        throws Exception
-    {
-        LoginCredentials cred = new LoginCredentials(args);
```
Comment (sbesson) [inline src/main/java/com/example/SimpleConnection.java]: Is it also worth adding a caveat to the Javadoc of https://github.com/openmicroscopy/openmicroscopy/blob/v5.4.0-m2/components/blitz/src/omero/gateway/LoginCredentials.java#L76?

File: src/main/java/com/example/SimpleConnection.java [hunk 2]
```diff
@@ -92,10 +78,31 @@ private void loadProjects()
...
+                if (tmp[0].trim().equals("--omero.port"))
+                    port = Integer.parseInt(tmp[1].trim());
+                if (tmp[0].trim().equals("--omero.user"))
+                    user = tmp[1].trim();
+                if (tmp[0].trim().equals("--omero.pass"))
+                    password = tmp[1].trim();
+            }
+
+            if (host != null && user != null && password != null) {
+                client.connect(host, port, user, password);
```
Comment (sbesson) [inline src/main/java/com/example/SimpleConnection.java]: As highlighted in the initial thread, would it be worth adding a simple logging line indicating the connection happened successfully?

File: src/main/java/com/example/SimpleConnection.java (hunk 1 above)
Comment (dominikl) [inline src/main/java/com/example/SimpleConnection.java]: Definetly. Though about removing this constructor, but I think we use it for integration tests, etc. I'll add a warning to the javadoc.

File: src/main/java/com/example/SimpleConnection.java (hunk 2 above)
Comment (dominikl) [inline src/main/java/com/example/SimpleConnection.java]: 👍

**Comment by sbesson** [comment]
NB: as per http://lists.openmicroscopy.org.uk/pipermail/ome-users/2017-August/006618.html, this fix might not be needed anymore if the `#` characters are escaped.

**Comment by joshmoore** [comment]
I'd agree. There's nothing particular;y wrong with the parsing here though it does introduce some duplication, I assume. That being said, I haven't thought of a way to prevent the issue we saw, other than fully parsing the strings pass to `--omero.*` and detect `#` without escaping. It might be easier to just specify in the javadocs of the gateway that `#` must be escaped.

**Comment by jburel** [comment]
next step?

**Comment by dominikl** [comment]
Forgot about that PR. I'll close it and just adjust the javadoc of the gateway.

**Pull request closed without merging**